Libs/Mesh/MeshUtils.cpp

```diff
@@ -168,4 +170,70 @@ int MeshUtils::findReferenceMesh(std::vector<Mesh>& meshes)
   return std::distance(means.begin(), smallest);
 }
 
+Mesh MeshUtils::computeMeanNormals(std::vector<Mesh> &meshes)
+{
+  if (meshes.empty())
+    throw std::invalid_argument("No meshes provided to compute average normals");
+
+  auto num_normals = meshes[0].numPoints();
+  auto num_meshes = meshes.size();
+
+  // convert all normals from all meshes to spherical coords
+  std::vector<std::vector<Point3>> sphericals(num_normals, std::vector<Point3>(num_meshes));
+  for (int j = 0; j < num_meshes; j++)
+  {
+    if (meshes[j].numPoints() != num_normals)
+      throw std::invalid_argument("Input meshes do not all have the same number of points");
+
+    auto normals = meshes[j].getField<vtkDataArray>("Normals");
+
+    if (num_normals != normals->GetNumberOfTuples())
+      throw std::invalid_argument("Expected a normal for every point in mesh");
+    
+    for (int i = 0; i < num_normals; i++)
+    {
+      auto n = normals->GetTuple3(i);
+
+      // note: Utils::cartesian2spherical returns atypical (r, phi, theta)
+      Utils::cartesian2spherical(n, sphericals[i][j].GetDataPointer());
+    }
+  }
+
+  // prep data in 1d theta/phi arrays for averageThetaArc function
+  std::vector<std::vector<double>> phis(num_normals, std::vector<double>(num_meshes));
+  std::vector<std::vector<double>> thetas(num_normals, std::vector<double>(num_meshes));
+  for (int i = 0; i < num_normals; i++)
+  {
+    for (int j = 0; j < num_meshes; j++)
+    {
+      phis[i][j] = sphericals[i][j][1];
+      thetas[i][j] = sphericals[i][j][2];
+    }
+  }
+
+  vtkSmartPointer<vtkDoubleArray> normals = vtkSmartPointer<vtkDoubleArray>::New();
+  normals->SetNumberOfComponents(3);
+  normals->SetNumberOfTuples(num_normals);
+  normals->SetName("MeanNormals");
+
+  // compute average value for collection of normals for all meshes
+  std::vector<Vector3> mean_normals(num_normals);
+  for (int i = 0; i < num_normals; i++)
+  {
+    Vector3 avg_spherical_normal = makeVector({1.0,
+                                               Utils::averageThetaArc(phis[i]),
+                                               Utils::averageThetaArc(thetas[i])});
+
+    // note: Utils::spherical2cartesian expects atypical (r, phi, theta)
+    Utils::spherical2cartesian(avg_spherical_normal.GetDataPointer(),
+                               mean_normals[i].GetDataPointer());
+
+    normals->SetTuple3(i, mean_normals[i][0], mean_normals[i][1], mean_normals[i][2]);
+  }
+
+  meshes[0].setField("MeanNormals", normals);
+
+  return meshes[0];
```

Comment:
If this function simply modifies a mesh, perhaps it should instead be a member of the Mesh class.
Otherwise it could return a field (until #935 is finished this would be a `vtkSmartPointer<vtkDoubleArray>`) or a new Mesh with that field, probably the former.
Recommended: keep it as a MeshUtils function and return a field.

Reply:
Done